org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -2277,23 +2277,13 @@ private String getAnnotationType(IJavaProject javaProject, List<String> annotati
 								}
 							}
 						}
+						if (!ProjectUtils.isGradleProject(javaProject.getProject()) && !ProjectUtils.isMavenProject(javaProject.getProject())) {
```

Comment:
I don't understand why we need to check if it's a Maven or Gradle project. We should strive to remain build agnostic as much as possible throughout jdt.ls

Reply:
The maven and gradle projects work fine. I have excluded them because of performance reasons.
Now, the PR skips this check.